app/views/products/_form.html.erb

```diff
@@ -27,6 +27,13 @@
     <%= form.text_field :image_url, id: :product_image_url %>
   </div>
 
+  <%= form.fields_for :images do |image_form| %>
+    <div class="field">
+      <%= image_form.label :url, 'Picture' %>
+      <%= image_form.file_field :url , multiple: true %>
```

Comment:
remove `multiple: true`
Instead of url (whicl will now rename as filepath) create a method say `def uploaded_image=(image_details)` and use the accessor here